aiida/common/log.py

```diff
@@ -212,6 +212,27 @@ def override_log_level(level=logging.CRITICAL):
         logging.disable(level=logging.NOTSET)
 
 
+@contextmanager
+def override_log_formatter_context(fmt: str):
+    """Temporarily use a different formatter for all handlers.
+
+    NOTE: One can _only_ set `fmt` (not `datefmt` or `style`).
+    Be aware! This may fail if the number of handlers is changed within the decorated function/method.
+    """
+    temp_formatter = logging.Formatter(fmt=fmt)
+    cached_formatters = []
```

Comment:
could use list comprehension here

Reply:
yep will do (I just copy/pasted the code from `override_log_formatter`)